weatherflow2mqtt/helpers.py

```diff
@@ -988,3 +993,52 @@ def fog_probability(self, solar_elevation, wind_speed, humidity, dew_point, air_
         fog_probability = round(fog)
 
         return fog_probability
+
+# *** Work in Progress ***
+    def snow_probability(self, air_temperature, freezing_level, cloud_base, dew_point, wet_bulb, station_height, is_metric: bool = None):
+        """ Return probability of snow in percent (Max of 80% calculated probability).
+        Input:
+            Air Temperature (metric)
+            Freezing Level (imperial or metric)
+            Cloud Base (imperial or metric)
+            Dew Point (metric)
+            Wet Bulb (imperial or metric))
+            Station Height (metric)
+        Where:
+            dptt is Dew Point plus Air Temperature
+            snow_prob is the probability of snow
+            snow_probability is the returned percentage of the probability of snow rounded (Max of 80%)
+        """
+        if (
+            air_temperature is None
+            or freezeing_level is None
+            or cloud_base is None
+            or dew_point is None
+            or wet_bulb is None
+        ):
+            return None
+
+        if is_metric is None:
+            is_metric = self.unit_system != UNITS_IMPERIAL
+
+        if is_metric:
+            #Do nothing if metric
+        else:
+            air_temperature = air_temperature
+            freezing_level = freezing_level / 3.28
+            cloud_base = cloud_base / 3.28
+            dew_point = dew_point
+            wet_bulb = (wet_bulb - 32) / 1.8
+            station_height = station_height
+        
+        snow_line = freezing_level - 228.6 # 750 ft / 228.6 m, snow line can vary in distance from freezing line
+        dptt = dew_point + temperature
+        
+        if ((air_temperature <= 2.1) && (snow_line <= station_height) && (dew_point <= 1) && (wet_bulb <= 2.1) && (freezing_line <= cloud_base) && (air_temperature >= -40)):
```

Comment:
You use a variable `freezing_line`, which is not supplied to the formula, we only have `freezing_level`. Should it be changed or where should the value come from?

Don't make any changes in this PR, as I have already merged it and fixed some minor issues, plus simplified the is_metric routine. 
`&&` in Python is `and` 😁

Reply:
I now assumed that `freezing_line` is the same as `freezing_level`, so I used that value in the formula.